extension/src/extension/background.ts

```diff
@@ -1,24 +1,20 @@
 import wretch from 'wretch';
-import { MsgType } from './types';
+import { MLISRequest, MLISResponse } from './types';
 
-
-var textToSummarize: string | null;
+const apiRoot: string = "https://aacxuouv1m.execute-api.us-east-1.amazonaws.com/default";
+const summarizationEndpoint: string = "/summarize";
 
 try {
     chrome.runtime.onMessage.addListener(
-        function (request, _, sendResponse) {
-            if (request.type === MsgType.PageContent) {
-                textToSummarize = request.text;
-            }
-            else if (request.type === MsgType.PopUpInit) {
-                if (textToSummarize != null) {
-                    sendResponse({
-                        textToSummarize: textToSummarize,
-                        // TODO: read article title from contentScript.ts & replace placeholder title
-                        articleTitle: "'Article Title Placeholder'"
-                    });
-                }
-            }
+        function (request: MLISRequest, _, sendResponse: (response: MLISResponse) => void) {
+            console.log("[Yubaba] Waiting for summarization process...")
+            wretch(apiRoot + summarizationEndpoint)
+                .options({ mode: "cors" })
+                .post({ url: request.url })
+                .json((mlisResponse: MLISResponse) => {
+                    console.log(mlisResponse)
+                    sendResponse(mlisResponse)
```

Comment:
I think we should add a status code check. 
`200 => sendResponse`
`500 => show error page`
`2xx => show message ` (ex. the article is not long enough to summarize)

Reply:
UI component for showing different messages can be done in a separate PR

Reply:
`wretch` can distinguish error codes but not successful codes.
So, I used `.error` to handle errors and `.res` to handle all successful responses👍